### ISSUE TITLE
Adding an icon for a certifier's credential.

### ISSUE THREAD

**Issue by mattgarrish** [w3c/epub-specs#1434]
Certifiers may want to include, or link out to, an icon for a credential.

See the discussion starting at https://github.com/w3c/epub-specs/issues/1410#issuecomment-724089441

**Comment by dauwhe** [comment]
Would such an icon be included in the package, with the expectation that a retailer would find the image in the package, extract it, and display it on the retailer website or in the user's library? That seems... extreme. 

Providing resources that are not part of the publication itself is problematic. Let's not forget all the chaos we've endured over cover images—I don't ever want to hear the term "logical cover" ever again.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2020-12-10-epub-a11y#section4) on 2020-12-10)

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Adding an icon for a certifier's credential 

_See github issue [#1434](https://github.com/w3c/epub-specs/issues/1434)._




**Charles LaPierre:** with GCA, we have 3 things: 1. GCA as text string, 2. Url to point to the page of the credential, 3. An icon or badge for certification.  
… potentially that badge could go somewhere inside the epub content, e.g. on the jacket.  
… and then the text string could be the alt-text for the badge.  
… we had this come up with VitalSource.  
… but we have to give them the notice to surface this info out of band.  
… it would be useful if this information could come from directly inside the epub.  

**George Kerscher:** what about if we create a new piece of metadata that is just a link to the badge?.  

**Charles LaPierre:** instead of having it just inside the epub?.  

**George Kerscher:** yes.  
… and then we could use the same refines technique.  

**Avneesh Singh:** where is the Url going right now Charles?.  

**Charles LaPierre:** right now it is going to a webpage that has the certification, with the publisher name, and the badge, etc..  
… but VitalSource doesn't want to have to go out to the webpage and look for where the badge is, etc..  

**Bill Kasdorf:** i'm open to these options, but from a production point of view, is that a GCA publisher would just put the badge on the copyright page.  
… so that badge might be in the book anyway.  

**Avneesh Singh:** Also, we've had issues with external content before, like when the cover was not part of the book.  
… if we are moving towards a direction where the badge is already part of the content of the book, and then we just point to it, then there is no problem.  
… but if the badge is just somewhere in the metadata, then this is more of an issue.  

**Ben Schroeter:** in general, i want to avoid tailoring metadata towards a specific use case.  
… the spec should be agnostic.  
… Bill, I don't think your suggestion is one that we're going to see come to fruition.  

**Bill Kasdorf:** ...and the a purchaser would have to have the book before the logo would be seen before they could but that logo on.  

**Ben Schroeter:** and plus, there are going to be non-accessible versions of the book too... like the PDF, or the print.  

**Bill Kasdorf:** right, i see how that is an issue.  

**Laura Brady:** As a publisher, we do put that GCA statement in all our books.  

**Charles LaPierre:** But the metadata in an epub version is specific to that version, right? It wouldn't be brought over to a PDF version.  
… so having the metadata pointing to an external image, or somewhere inside the epub are both options that seem equivalent to me.  
… why would one be a problem, and the other not?.  

**Avneesh Singh:** Okay, let's explore this more in the github issue and come back later for a resolution.  

**Charles LaPierre:** we could also do both: have 1 badge inside the content, and 1 in an external point URL.  

**Avneesh Singh:** Another point is that content in the epub will remain static over time. If there is a badge refresh, a URL method would keep everything consistent.  

**Charles LaPierre:** right, but then internet connection issues will impact an external file method....  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2020-12-10-epub-a11y#section4) on 2020-12-10)

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Adding an icon for a certifier's credential 

_See github issue [#1434](https://github.com/w3c/epub-specs/issues/1434)._




**Charles LaPierre:** with GCA, we have 3 things: 1. GCA as text string, 2. Url to point to the page of the credential, 3. An icon or badge for certification.  
… potentially that badge could go somewhere inside the epub content, e.g. on the jacket.  
… and then the text string could be the alt-text for the badge.  
… we had this come up with VitalSource.  
… but we have to give them the notice to surface this info out of band.  
… it would be useful if this information could come from directly inside the epub.  

**George Kerscher:** what about if we create a new piece of metadata that is just a link to the badge?.  

**Charles LaPierre:** instead of having it just inside the epub?.  

**George Kerscher:** yes.  
… and then we could use the same refines technique.  

**Avneesh Singh:** where is the Url going right now Charles?.  

**Charles LaPierre:** right now it is going to a webpage that has the certification, with the publisher name, and the badge, etc..  
… but VitalSource doesn't want to have to go out to the webpage and look for where the badge is, etc..  

**Bill Kasdorf:** i'm open to these options, but from a production point of view, is that a GCA publisher would just put the badge on the copyright page.  
… so that badge might be in the book anyway.  

**Avneesh Singh:** Also, we've had issues with external content before, like when the cover was not part of the book.  
… if we are moving towards a direction where the badge is already part of the content of the book, and then we just point to it, then there is no problem.  
… but if the badge is just somewhere in the metadata, then this is more of an issue.  

**Ben Schroeter:** in general, i want to avoid tailoring metadata towards a specific use case.  
… the spec should be agnostic.  
… Bill, I don't think your suggestion is one that we're going to see come to fruition.  

**Bill Kasdorf:** ...and the a purchaser would have to have the book before the logo would be seen before they could but that logo on.  

**Ben Schroeter:** and plus, there are going to be non-accessible versions of the book too... like the PDF, or the print.  

**Bill Kasdorf:** right, i see how that is an issue.  

**Laura Brady:** As a publisher, we do put that GCA statement in all our books.  

**Charles LaPierre:** But the metadata in an epub version is specific to that version, right? It wouldn't be brought over to a PDF version.  
… so having the metadata pointing to an external image, or somewhere inside the epub are both options that seem equivalent to me.  
… why would one be a problem, and the other not?.  

**Avneesh Singh:** Okay, let's explore this more in the github issue and come back later for a resolution.  

**Charles LaPierre:** we could also do both: have 1 badge inside the content, and 1 in an external point URL.  

**Avneesh Singh:** Another point is that content in the epub will remain static over time. If there is a badge refresh, a URL method would keep everything consistent.  

**Charles LaPierre:** right, but then internet connection issues will impact an external file method....  

</details>

**Comment by gregoriopellegrino** [comment]
For the Italian market and the Fondazione LIA this is of great interest. At the moment some publishers insert the logo in the copyright page, but without any particular semantic value.

**Comment by avneeshsingh** [comment]
We sort of stalled on this issue.
@clapierre and @gregoriopellegrino, it would be good to have inputs / feedback from retailers / distributors like VitalSource, to know how do they intend to use the icon, and is it important for them to have link to ICON in the publication or icon embedded in the publication.

**Comment by gregoriopellegrino** [comment]
In Italy Italian retailers are displaying LIA logo in the book page to show the accessibility certification, but I think global resellers will be reluctant to display it.

**Comment by clapierre** [comment]
Vitalsource is already showing our GCA Certification Icon in their all their stores if the book has been certified by us.

Here is an [example of a book at our bookstore](https://benetechaccessiblebooks.vitalsource.com/products/the-good-retirement-guide-2021-v9781789667608), if you scroll down to the Accessibility Tab and click on it you will bee the Global Certified Accessible by Benetech logo.  

Since they couldn't get this information from the EPUB itself we had to send VitalSource directly our Logo and when they detect a book with our "certifierCredential" matches they use the logo.  This is not a great practice as now, we have to do this with Redshelf next as well and any other Library / Bookstore around the world.

We need to make this easier for bookstores and libraries to get this from the EPUB itself which in the metadata can link to the image ideally inside the EPUB but there may be cases when a URL pointing outside the EPUB to a downloadable image may be requested as an option.  On this point of having the logo image as a URL metadata pointing outside the EPUB I would think that would be more work for the bookstore & libraries to implement but wondering with others think.

Once we have this logo accessible through the metadata we can then update our documentation [User Experience Guide for Displaying Accessibility Metadata](https://w3c.github.io/publ-a11y/UX-Guide-Metadata/principles/) on how this new metadata can be used to display the certifierCredentials and the certifierLogo.

**Comment by mattgarrish** [comment]
> We need to make this easier for bookstores and libraries to get this from the EPUB itself which in the metadata can link to the image ideally inside the EPUB but there may be cases when a URL pointing outside the EPUB to a downloadable image may be requested as an option.

Is it realistic they would do this, though? This seems like it could be a security minefield.

What's to stop someone bent on mischief from putting in a fake GCA certification with a link to a dirty picture or an image with a security exploit?

I think it would be very helpful to know for sure that vendors want this as the solution. There are other options we could consider, like keeping a registry of certifiers logos somewhere so that there's an authoritative but public list of what logos to associate with whose credentials.

**Comment by clapierre** [comment]
That works as well Matt, I am fine if there is a registration of sorts where the "certifierCredential" can be looked up to retrieve the "certificationLogo" but I also think the publisher may want to have the certificationLogo embedded in the EPUB so that it can be displayed on the cover / back cover or title page potentially.

To your point if the publisher was actually pointing to the correct Logo on the web and someone hacked that website that could be as you point out a bad situation, so I totally agree that wouldn't be a good option.  Even a registry of sorts could get hacked but having the image logo inside the EPUB seems like the safest option to me.

**Comment by mattgarrish** [comment]
> I also think the publisher may want to have the certificationLogo embedded in the EPUB so that it can be displayed on the cover / back cover or title page potentially.

You don't need to use a link element to do this, though. You just embed it in the appropriate content document, or link out to it. Any mischief falls on the publisher as it's their content.

The concern for a third party is displaying something in their own bookstore or on their reading system that they can't trust. There's no way to control who can specify a credential or what image they can use.

The only way I'd see a link working is if it always referenced an official URL for an image, so the link would be more like an identifier. A vendor could obtain a safe copy of the image and display it if the URL is what they expect.

But this seems redundant to performing the same sort of operation on the credential itself. So, again, this is where it would be helpful to hear from possible implementers what model they'd expect to use.

> Even a registry of sorts could get hacked

Sure, everything can be hacked, which is why if I were implementing this I'd be wary about trusting anything except what I've obtained and checked myself.

**Comment by mattgarrish** [comment]
And to be clear, I wasn't suggesting we host logos in a registry. If we create one, it should only identify the credential and point people to where to obtain the official logos to display. I'd only trust the certifier's web site to have safe images.

**Comment by mattgarrish** [comment]
Looking at this again, do we need to do anything for this issue?

If you really want to include an icon, you could always use the link element with an image media type:

```
<link rel="a11y:certifierCredential" refines="..." href="https://example.com/credential-image.jpg" media-type="image/jpeg"/>
<meta property="a11y:certifierCredential" refines="...">Credential Name</meta>
```

**Comment by mattgarrish** [comment]
Closing this issue for the same reason given in https://github.com/w3c/epub-specs/issues/1590#issuecomment-838291042

If this doesn't work, we'll at least have a better idea of what is needed.

**Comment by clapierre** [comment]
I think in the Techniques Document we should have an example and consistent way so that a Library or Bookstore knows where to look in the Metadata for this Logo thats either imbedded or provided with a URI.

**Comment by mattgarrish** [comment]
Do we know that anyone is currently using icons embedded in an EPUB, or that they are committed to doing so?

We should ensure that the practices we recommend have uptake before suggesting to people to use them. This might be something to first explore in the document on how to display the metadata, for example.
